### PR TITLE
Fix (some) non-tuple index warnings

### PR DESCRIPTION
**Pull request by ev-br**
As a companion to gh-8879, fix the non-tuple index warnings in `signal`, `interpolate` and `stats`.

### CODE REVIEW

**Review by larsoner — APPROVED** [review]
Changes look reasonable to me, +1 for merge

File: scipy/interpolate/tests/test_interpolate.py [hunk 1]
```diff
@@ -2546,7 +2546,7 @@ def __init__(self, shape):
         self._v = np.arange(np.prod(shape)).reshape(shape)
 
     def __getitem__(self, idx):
-        return self._v[idx]
+        return self._v[tuple(idx)]
```
Comment (eric-wieser) [inline scipy/interpolate/tests/test_interpolate.py]: Might make more sense to let the error propagate here, and fix it at the caller instead

File: scipy/signal/signaltools.py [hunk 2]
```diff
@@ -488,7 +488,7 @@ def _reverse_and_conj(x):
     Reverse array `x` in all dimensions and perform the complex conjugate
     """
     reverse = [slice(None, None, -1)] * x.ndim
```
Comment (eric-wieser) [inline scipy/signal/signaltools.py]: Could just change this one to a tuple literal rather than a list

Comment (ev-br) [reply scipy/signal/signaltools.py]: OK, done

File: scipy/interpolate/tests/test_interpolate.py (hunk 1 above)
Comment (ev-br) [reply scipy/interpolate/tests/test_interpolate.py]: Actually, I failed to trigger this warning with the numpy HEAD, so I've just reverted this change.

**Review by eric-wieser — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks @ev-br

**Pull request merged**